test/upgrade/prober/wathola/event/services.go

```diff
@@ -91,7 +95,11 @@ func (f *finishedStore) RegisterFinished(finished *Finished) {
 	time.Sleep(d)
 	receivedEvents := f.steps.Count()
 	if receivedEvents != finished.Count {
-		f.errors.throw("expecting to have %v unique events received, "+
+		throwMethod := f.errors.throwMissing
+		if receivedEvents > finished.Count {
+			throwMethod = f.errors.throwUnexpected
+		}
+		throwMethod("expecting to have %v unique events received, "+
```

Comment:
I was telling before that I maybe it's not a good idea to throw this missing error here at all. We have error for each missing event. 

For ex.: 

if we miss 1 event we will have 2 errors:
 * expected to have 123 unique events received, but received 122 unique events
 * event 54 should be received once, but was received 0 times

But it's also good to have the total number of events reported. But, we can easily have that with your second PR. So I'd say lets not add here this additional missing error.

Reply:
I use unexpected exclusively here.